File: Archiver/app_code/clsNetworkFileListener.vb
Imports System
Imports System.IO

Public Class clsNetworkFileListener
    Public networkDir As String = ""

    Private Sub SetListener()

        Dim watcher As FileSystemWatcher = New FileSystemWatcher(networkDir, "*.*")
        AddHandler watcher.Created, AddressOf OnCreated
        watcher.IncludeSubdirectories = True
        watcher.InternalBufferSize = 25 * 4096
        watcher.NotifyFilter = NotifyFilters.FileName
        watcher.EnableRaisingEvents = True

        MessageBox.Show("Close to End Listener")

        'While Console.ReadLine() <> "x"
        'End While

        watcher.Dispose()
    End Sub

    Private Shared Sub OnCreated(ByVal source As Object, ByVal e As FileSystemEventArgs)
        Console.WriteLine("File: " & e.FullPath & " " & e.ChangeType)
    End Sub

End Class

File: Archiver/app_code/clsNetworkFileListener.cs
using global::System;
using global::System.IO;
using System.Windows.Forms;
using MODI;

namespace EcmArchiver
{
    public class clsNetworkFileListener
    {
        public string networkDir = "";

        private void SetListener()
        {
            var watcher = new FileSystemWatcher(networkDir, "*.*");
            watcher.Created += OnCreated;
            watcher.IncludeSubdirectories = true;
            watcher.InternalBufferSize = 25 * 4096;
            watcher.NotifyFilter = NotifyFilters.FileName;
            watcher.EnableRaisingEvents = true;
            MessageBox.Show("Close to End Listener");

            // While Console.ReadLine() <> "x"
            // End While

            watcher.Dispose();
        }

        private static void OnCreated(object source, FileSystemEventArgs e)
        {
            Console.WriteLine("File: " + e.FullPath + " " + ((int)e.ChangeType).ToString());
        }
    }
}